tpm/tss2/signer.go

```diff
@@ -130,13 +136,30 @@ func CreateSigner(rw io.ReadWriter, key *TPMKey) (*Signer, error) {
 // Notice: This API is EXPERIMENTAL and may be changed or removed in a later
 // release.
 func (s *Signer) SetSRKTemplate(p tpm2.Public) {
+	s.m.Lock()
 	s.srkTemplate = p
+	s.m.Unlock()
+}
+
+// SetTPM allows to change the TPM channel. This operation is useful if the
+// channel set in [CreateSigner] is closed and opened again before calling [Signer.Sign].
+//
+// # Experimental
+//
+// Notice: This API is EXPERIMENTAL and may be changed or removed in a later
+// release.
+func (s *Signer) SetTPM(rw io.ReadWriter) {
```

Comment:
Maybe call this `SetCommandChannel` instead? Although it's not entirely the same as `attest.CommandChannel`, but it sort of has that meaning in this context.

Reply:
Fixed with 3f48274